metadata-ingestion/src/datahub/ingestion/source/tableau_common.py

```diff
@@ -573,17 +578,118 @@ def get_fully_qualified_table_name(
     return fully_qualified_table_name
 
 
-def get_platform_instance(
-    platform: str, platform_instance_map: Optional[Dict[str, str]]
-) -> Optional[str]:
-    if platform_instance_map is not None and platform in platform_instance_map.keys():
-        return platform_instance_map[platform]
+@dataclass
+class TableauUpstreamReference:
+    database: Optional[str]
+    schema: Optional[str]
+    table: str
+
+    connection_type: str
+
+    @classmethod
+    def create(
```

Comment:
Meant to replicate logic in `get_upstream_tables`. Pulls values out of `table` dict, parses schema from `full_name` and `default_schema_map` (and now parses `database` too), and does this `schema in table` check for redshift